### ISSUE TITLE
il.org.spartan.spartanizer.engine -> makeAST.java & il.org.spartan.spartanizer.ast.navigate -> cantTip.java (Assignment 2) Code Coverage

### ISSUE THREAD

**Issue by yosefraisman** [SpartanRefactoring/Spartanizer#822]
Increasing code coverage of il.org.spartan.plugin/src/main/java/il/org/spartan/spartanizer/engine/makeAST.java #799. Current coverage is 47.4%, please don't choose this file.

**Comment by yossigil** [comment]
#616: Good job so far.

**Comment by AmitOhayon** [comment]
@yossigil 
during our work we found a bug! are we allowed to fix it?

**Comment by AmitOhayon** [comment]
@yossigil 
Many methods use IFile as parameter, which apparently represent files from the workspace (?)
while trying to test one of this methods, I got an "IllegalStateException" from ResourcesPlugin.getWorkspace()
google comes up with this: http://stackoverflow.com/questions/962364/workspace-is-closed-exception-when-trying-to-get-the-workspace-through-resourc
I don't want to mess with files such as Manifest.MF without knowing what they are.
Bottom Line: should we cover these methods? what can we do to overcome this exception?

**Comment by yossigil** [comment]
Use mockito. Search the code for mocking examples

**Comment by AmitOhayon** [comment]
@yossigil 
i suspect we cant use mockito here because we dont know which methods of IFile are invoked by JavaCore.createCompilationUnitFrom(IFile) so we cant implement them

**Comment by yossigil** [comment]
I think you can do a bit better. Recall cs 236703, :+1: 
but if it is too tough, perhaps you should move elsewhere

**Comment by AmitOhayon** [comment]
@yossigil 
we really tried for hours, but for no avail. we pushed our working tests, and with your permission we would like to work on ast/navigate/cantTip.java as well

**Comment by yossigil** [comment]
NP :+1:

**Comment by orimarco** [comment]
@tonylekhtman @yosefraisman @AmitOhayon 
You have a failing test - `returnsNullOnIOException`

**Comment by AmitOhayon** [comment]
@orimarco 
I'm not seeing it...
![screenshot from 2016-11-15 10-53-17](https://cloud.githubusercontent.com/assets/1254412/20299041/c4e44764-ab21-11e6-8c5c-874838e0f9b2.png)

**Comment by yossigil** [comment]
as of 42562719d57ec9b359a9c16ad55475dd843bc7d9 all tests are clean on my side

**Comment by yossigil** [comment]
ISSUE OPEN TWO DAYS AFTER MILESTONE>

**Comment by yosefraisman** [comment]
We did close it in time, but it was reopened by @orimarco

**Comment by yossigil** [comment]
@yosefraisman :-1: it was opened becasue you had a failing test. You were suppposed to take care of it IMMEDIATELY. Reopening an issue means you had problems. You cannot just sit there and ignore these:

**Comment by yosefraisman** [comment]
We did reply to Ori, within one hour of him reopening the issue, saying none of our tests actually were failing. I thought it doesn't make sense for us to close it again before we find out why it was originally reopened, and get confirmation that it can be closed (since we made no changes to the code).

**Comment by yossigil** [comment]
One response could have been:
```
As per Yossi's message: "as of 4256271 all tests are clean on my side", we close this issue, problem not confirmed. Reopen if confirmed.".
```
Or 
````
Checking now, trying to reporduce the bug
```
Or, 
```
Checking COMMIT1, COMMIT2, ...., the problem is gone, closing
```
etc

**Comment by yosefraisman** [comment]
Thanks for your feedback, I'll bear it in mind in the future.

**Comment by yossigil** [comment]
:+1:
